docs/usage/management_commands.rst

```diff
@@ -1,27 +1,70 @@
 Management Commands
 ===================
 
-migrate_with_timeouts
----------------------
+``migrate_with_timeouts``
+-------------------------
 
 Runs database migrations with timeouts.
 
 This command wraps the :ref:`apply_timeouts() <apply_timeouts>` around
-Django's ``migrate`` command, exposing two extra arguments:
+Django's ``migrate`` command, exposing extra arguments to handle retries with
+backoff propagation.
 
-- ``--lock-timeout-in-ms``
-- ``--statement-timeout-in-ms``
++++++++++++++++++++++++
+lock_timeout arguments:
++++++++++++++++++++++++
 
- .. important::
+* ``--lock-timeout-in-ms``: Sets the lock_timeout value for the migration.
+* ``--lock-timeout-max-retries``: How many times to retry after getting a
+  lock_timeout. Defaults to zero, which means no retries.
+* ``--lock-timeout-retry-exp``: The exponential to use for exponential backoff
+  retry if retries are enabled. Defaults to 2.
+* ``--lock-timeout-retry-min-wait-in-ms``: minimum amount of time to wait
+  between lock_timeout retries in milliseconds. Defaults to 3s.
+* ``--lock-timeout-retry-max-wait-in-ms``: Maximum amount of time to wait
+  between lock_timeout retries in milliseconds. Defaults to 10s.
 
-   Both arguments are optional, but at least one must be provided!
+++++++++++++++++++++++++++++
+statement_timeout arguments:
+++++++++++++++++++++++++++++
 
-These arguments will set the value of Postgres' ``lock_timeout`` and
-``statement_timeout`` for the duration of the migration.
+* ``--statement-timeout-in-ms``: Sets statement_timeout for the migration.
+* ``--statement-timeout-max-retries``: How many times to retry after getting a
+  statement_timeout. Defaults to zero, which means no retries.
+* ``--statement-timeout-retry-exp``: The exponential to use for exponential
+  backoff retry if retries are enabled. Defaults to 2.
+* ``--statement-timeout-retry-min-wait-in-ms``: minimum amount of time to wait
+  between statement timeout retries in milliseconds. Defaults to 3s.
+* ``--statement-timeout-retry-max-wait-in-ms``: Maximum amount of time to wait
+  between statement timeout retries in milliseconds. Defaults to 10s.
 
-==========
++++++++++++++++
+retry callback:
++++++++++++++++
+
+* ``--retry-callback--path``: Sets a callback to be called after a timeout
```

Comment:
copy-paste: remove double hyphen here: `...callback--path`

Reply:
Refer to https://github.com/kraken-tech/django-pg-migration-tools/pull/20#discussion_r1760473678